trino-s3-proxy/src/test/java/io/trino/s3/proxy/server/testing/TestingSecurityFacade.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.s3.proxy.server.testing;
+
+import io.trino.s3.proxy.server.rest.ParsedS3Request;
+import io.trino.s3.proxy.server.security.SecurityFacade;
+import io.trino.s3.proxy.server.security.SecurityFacadeProvider;
+import io.trino.s3.proxy.server.security.SecurityResponse;
+import jakarta.ws.rs.WebApplicationException;
+
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static java.util.Objects.requireNonNull;
+
+public class TestingSecurityFacade
+        implements SecurityFacade, SecurityFacadeProvider
+{
+    private final AtomicReference<SecurityFacade> delegate = new AtomicReference<>((lowercaseHttpMethod, lowercaseAction, region, bucket, key) -> SecurityResponse.DEFAULT);
+
+    @Override
+    public SecurityFacade securityFacadeForRequest(ParsedS3Request request)
+            throws WebApplicationException
+    {
+        return this;
+    }
+
+    public SecurityFacade setDelegate(SecurityFacade delegate)
+    {
+        return this.delegate.getAndSet(requireNonNull(delegate, "delegate is null"));
```

Comment:
Is there any occasion where we'd need to return the old `SecurityFacade` delegate?

Reply:
You'd have a test that wants to set the delegate and then reset it to whatever it was before. This way you can nest usages of the delegate.

Reply:
But given that we have a getter it's not necessary. I'll change.

Reply:
I'm not against this behaviour, but I think that a method called `setDelegate` returning the _old_ delegate is confusing, particularly given that the pattern we follow in most builders is that a `set` method returns the builder itself or the object being constructed, not its _previous_ state.

If you prefer to keep this for testing, then maybe we can call this `replaceDelegate`?

Alternatively, we can make this return nothing

Reply:
I'm OK with returning `void`. I'll change it.